### DISCUSSION TITLE
Automatic Getter function for dynamic arrays in struct

### DISCUSSION THREAD

**Discussion by 18121A05L2** [Cyfrin/foundry-full-course-cu#2955] [Q&A]
```
    struct NetworkConfig {
        address[] tokenAddresses;
        address[] priceFeedAddresses;
    }

    NetworkConfig public networkConfig;
```
    
    i was facing this issue - The struct has all its members omitted, therefore the getter cannot return any values.
    
    i know we need to create separate getter functions , but why solidity won't create getter function by default as reading data 
    from storage ( blockchin ) is free outside of the EVM 
    
    it just need to loop through the entire array and just returning , why we have gas here ( i have checked in google ) ?

**Comment by tcrin** [comment]
Hi @18121A05L2,

### ❓ 1. I was facing this issue - The struct has all its members omitted, therefore the getter cannot return any values.
✅ Solidity will skip generating getters for mappings and arrays inside structs (with the exception of byte arrays).

---

### ❓ 2. I know we need to create separate getter functions, but why won't Solidity create getter functions by default as reading data from storage (blockchain) is free outside of the EVM?
✅ It is not possible to access elements in a mapping without specifying a specific key.  
For arrays, if the specific element is not identified, the EVM will have to loop through the entire array to return the result. Iterating through each element in the array consumes computational resources and leads to gas costs.

---

### ❓ 3. It just needs to loop through the entire array and return the values, so why do we have gas costs here (I have checked on Google)?
✅ Gas fees help keep the Ethereum network secure. By requiring a fee for every computation executed on the network, gas prevents bad actors from spamming the network.

---
The answer is based on my knowledge, and if there are any mistakes, I welcome any suggestions or corrections from everyone

**Reply by EngrPips** [reply]
Great heads-up here.
